api/v1/bmc.proto

```diff
@@ -38,3 +41,43 @@ message ResetRequest {
 message ResetResponse {
     string task_id = 1;
 }
+
+message UserCreds {
+    string username = 1;
+    string password = 2;
+    enum Role {
+        admin = 0;
```

Comment:
According to https://developers.google.com/protocol-buffers/docs/style#enums 0th enum should be left as unspecified. This seems like a good idea to keep the 0th value as unspecified.

Also should be prefixed instead of in the message and the values should be prefixed and all caps. But these seem like more nits than good ideas overall.